nixos/modules/services/video/wivrn.nix

```diff
@@ -0,0 +1,130 @@
+{ config, pkgs, lib, ... }:
+let
+  inherit (lib) mkAliasOptionModule mkIf mkEnableOption mkPackageOption mkOption mkDefault optionalString getExe' maintainers;
+  cfg = config.services.wivrn;
+  configFormat = pkgs.formats.json { };
+  configFile = configFormat.generate "config.json" cfg.config.json;
+in
+{
+  options = {
+    services.wivrn = {
+      enable = mkEnableOption "WiVRn, an OpenXR streaming application";
+
+      package = mkPackageOption pkgs "wivrn" { };
+
+      openFirewall = mkEnableOption "the default ports in the firewall for the WiVRn server";
+
+      defaultRuntime = mkEnableOption ''
+        WiVRn Monado as the default OpenXR runtime on the system. The config can be found at `/etc/xdg/openxr/1/active_runtime.json`.
+
+        Note that applications can bypass this option by setting an active
+        runtime in a writable XDG_CONFIG_DIRS location like `~/.config`
+      '' // { default = true; };
+
+      highPriority = mkEnableOption "high priority capability for wivrn-server" // { default = true; };
+
+      config = {
+        enable = mkEnableOption "configuration for WiVRn";
+        json = mkOption {
+          type = configFormat.type;
+          description = ''
+            Configuration for WiVRn. The attributes are serialized to JSON in config.json.
+            See https://github.com/Meumeu/WiVRn/blob/master/docs/configuration.md
+          '';
+          default = {
+            scale = 0.8;
+            bitrate = 50000000;
+            encoders = [
+              {
+                encoder = "nvenc";
+                codec = "h265";
```

Comment:
Since NVIDIA's stuff is proprietary and only works with an NVIDIA GPU, it can't be the default. Additionally, the h265 codec itself is proprietary, so it _also_ can't be the default.

Almost all CPUs post-2011 have hardware h264 acceleration, so h264 + VAAPI makes sense as an open-source engine / encoder combination that will work on _most_ systems.

Reply:
It's what's set to default upstream so that's what I put here, but we can always change it. By default the config option is not even enabled.